TFT/src/User/API/extend.c

```diff
@@ -202,31 +212,28 @@ bool FIL_IsRunout(void)
         return false;
     }
   }
+
   return false;
 }
 
 void loopBackEndFILRunoutDetect(void)
 {
-  if (!(infoSettings.runout & 1))   // Filament runout turn off
-    return;
-  if (!isPrinting() || isPaused())  // Check for No printing or printing paused before reading sensor
-    return;
-  if (!FIL_IsRunout())              // Filament not runout yet, need constant scanning to filter interference
+  if (!(infoSettings.runout & 1))  // Filament runout turn off
```

Comment:
Removing the printing/pause check at line 212 will bring back #2024 bug. The sensor state should not be read during pause/resume procedure. This will result in incorrect positions which may cause damage to the nozzle or the print if the nozzle is lowered in to the print.

Reply:
Hmm no. After you got the first filament runout message, the print is paused and the nozzle is parked. Then you cannot resume the print until the runout is fixed (e.g. filament is changed). This PR simply avoid you can resume the print if runout is still there (bug 2024 is avoided)

Reply:
Does the logic work for smart filament sensors also?

Reply:
it remained the same as the old logic (previous to your PR). For SFS, the runout value is used just to detect a transition (the value is not even important). I'm going also to mount an SFS.
If you have already it and you could check it it will be appreciated

Reply:
I will do in a few hours with both kinds of sensors

Reply:
I installed the SFS on my printer and made some testing. The logic for SFS works as the previous logic. The runout value is handled in a different way than standard sensor. So it takes some time before triggering a new condition. But that's the current logic provided in FIL_SmartRunoutDetect() function that could maybe improved in a further PR

Reply:
Tested with both kind of sensors successfully. 

👍🏻